version/version.go

```diff
@@ -1,5 +1,24 @@
 package version
 
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
 // Version specifies the current version of Draupnir
 // This value is injected in at compile time (see the Makefile)
 var Version string
+
+// ParseSemver extracts the major minor and patch level versions from a version string.
+func ParseSemver(version string) (int, int, int, error) {
+	if !regexp.MustCompile("^\\d+\\.\\d+\\.\\d+$").Match([]byte(version)) {
+		return -1, -1, -1, fmt.Errorf("version was not a valid semver: %s", version)
+	}
+
+	mustAtoi := func(s string) int { i, _ := strconv.Atoi(s); return i }
+
+	components := strings.Split(version, ".")
```

Comment:
Could use `FindAllSubmatch` on the regex here, instead of splitting on dots. I don't mind much one way or the other though.

Reply:
Think I'm going to leave it as it is, though `FindAllSubmatch` is nice!